src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1226,12 +1227,18 @@ namespace winrt::TerminalApp::implementation
             // process until later, on another thread, after we've already
             // restored the CWD to its original value.
             auto newWorkingDirectory{ settings.StartingDirectory() };
-            if (newWorkingDirectory.size() == 0 || newWorkingDirectory.size() == 1 &&
-                                                       !(newWorkingDirectory[0] == L'~' || newWorkingDirectory[0] == L'/'))
-            { // We only want to resolve the new WD against the CWD if it doesn't look like a Linux path (see GH#592)
-                auto cwdString{ wil::GetCurrentDirectoryW<std::wstring>() };
+            const bool looksLikeLinux = newWorkingDirectory.size() == 1 &&
+                                        (newWorkingDirectory[0] == L'~' || newWorkingDirectory[0] == L'/');
+            // We only want to resolve the new WD against the CWD if it doesn't look like a Linux path (see GH#592)
+            if (!looksLikeLinux)
```

Comment:
We dropped the `size == 0` check - is that safe? :)

Reply:
Wait, something seems very wrong here.

The check on the left says...

* It is empty (why would we bother appending? Oh well, we used to)
* OR
   * It is one character long (???) and
   * It is **not** `~` or `/`

That doesn't make sense. Digging.

FWIW: I think this change broke `wt wsl`

Reply:
I feel like the condition you removed and the one you added aren't equivalent. I was trying to think it through. My brain isn't working.

Reply:
I'm not totally sure I understood the condition _before_ 😆

Reply:
Previously it was
```cpp
size == 0 || size == 1 && !(begin == L'~' || begin == L'/')
```
which is equivalent to
```cpp
size == 0 || (size == 1 && (begin != L'~' && begin != L'/'))
```

and now it is
```cpp
!(size == 1 && (begin == L'~' || begin == L'/'))`
```
which is equivalent to
```cpp
size != 1 || (begin != L'~' && begin != L'/')`
```
which seems wrong because it now overwrites `newWorkingDirectory` even if it already contains something.

I don't understand the old condition either though. I would've expected it to just say:
```cpp
size == 0 || (begin != L'~' && begin != L'/')
```
to support longer paths like `~/foo/bar`. I suppose I don't understand what that code does in general.

Reply:
Okay so it's starting to come back to me. 

The old code definitely didn't always append the startingDirectory to the CWD of the Terminal. It.... kinda looks like it only appended to the CWD if it was one character and _wasn't_ `~` or `/`. 

That was changed here: https://github.com/microsoft/terminal/blame/c98ec15144d2cc76c7332aea8bebcf1831201777/src/cascadia/TerminalApp/TerminalPage.cpp#L944-L945 (#12050)
```c++
if (newWorkingDirectory.size() == 0 || newWorkingDirectory.size() == 1 &&
                                       !(newWorkingDirectory[0] == L'~' || newWorkingDirectory[0] == L'/'))
```

Before that was:
https://github.com/microsoft/terminal/blame/4930508b8c9d2e1c50a20763d075de529ac9cf7d/src/cascadia/TerminalApp/TerminalPage.cpp#L943-L951 (#9223)
```c++
if (newWorkingDirectory.size() <= 1 ||
    !(newWorkingDirectory[0] == L'~' || newWorkingDirectory[0] == L'/'))
```

So basically #12050 just made us not do the appending when it was a single `~` or `/`, where before that we _would_. 

<hr>

But we **never** would append for any of:
* `./dev`
* `/dev`
* `~/dev`

Because for those:
* it's a relative path, so just leaving it be relative was fine! CreateProcess would just create it relative to our CWD, QED.
* It's an absolute path - `MangleStartingDirectoryForWSL` will do the right thing if it's a WSL though
* It's a relative path - `MangleStartingDirectoryForWSL` will do the right thing if it's a WSL though

So as of this PR, we actually _do_ need to change the logic, to append... basically always. UNLESS it's 
* `~/thing`
* `/thing`

Reply:
Hey look, I wrote some tests for this. Sorry that it's in `types/Utils`, you know how it be